Not Able to execute the the test command as per the getting started guide

Testing Babelfish drivers as per the getting started guide.But getting below error while testing:

command:

sudo docker exec -it bblfshd bblfshctl parse /opt/bblfsh/etc/examples/python.py

Status: Fatal
Elapsed: 5.000365287s
Errors:
	- timeout, all drivers are busy
pls help me if its an issue.

Thanks.

Got the same problem (Fedora 27):

### First try 
```bash
$> docker exec -it bblfshd bblfshctl parse /opt/bblfsh/etc/examples/python.py
Status: Fatal
Elapsed: 124.445374ms
Errors:
	- unexpected error: container_linux.go:296: starting container process caused "permission denied"
```
### Further tries
```bash
$> docker exec -it bblfshd bblfshctl parse /opt/bblfsh/etc/examples/python.py
Status: Fatal
Elapsed: 5.00086396s
Errors:
	- timeout, all drivers are busy
```

### Your request
```bash
$> docker logs bblfshd
time="2018-02-26T10:15:01Z" level=info msg="bblfshd version: v2.4.0 (build: 2018-02-21T18:11:17+0000)"
time="2018-02-26T10:15:01Z" level=info msg="initializing runtime at /var/lib/bblfshd"
time="2018-02-26T10:15:01Z" level=info msg="control server listening in /var/run/bblfshctl.sock (unix)"
time="2018-02-26T10:15:01Z" level=info msg="server listening in 0.0.0.0:9432 (tcp)"
time="2018-02-26T10:15:59Z" level=info msg="driver java installed "bblfsh/java-driver:latest""
time="2018-02-26T10:16:15Z" level=info msg="driver javascript installed "bblfsh/javascript-driver:latest""
time="2018-02-26T10:16:29Z" level=info msg="driver typescript installed "bblfsh/typescript-driver:latest""
time="2018-02-26T10:18:19Z" level=info msg="driver python installed "bblfsh/python-driver:latest""
time="2018-02-26T10:18:51Z" level=info msg="driver ruby installed "bblfsh/ruby-driver:latest""
time="2018-02-26T10:19:04Z" level=info msg="driver go installed "bblfsh/go-driver:latest""
time="2018-02-26T10:19:27Z" level=info msg="driver php installed "bblfsh/php-driver:latest""
time="2018-02-26T10:20:12Z" level=info msg="driver bash installed "bblfsh/bash-driver:latest""
panic: permission denied
goroutine 1 [running, locked to thread]:
github.com/bblfsh/bblfshd/runtime.Bootstrap()
	/go/src/github.com/bblfsh/bblfshd/runtime/runtime.go:198 +0x15d
main.init.0()
	/go/src/github.com/bblfsh/bblfshd/cli/bblfshd/main.go:61 +0x499
main.init()
	<autogenerated>:1 +0x7b
time="2018-02-26T10:22:11Z" level=error msg="error selecting pool: unexpected error: container_linux.go:296: starting container process caused "permission denied""
time="2018-02-26T10:22:11Z" level=error msg="request processed content 23 bytes, status Fatal" elapsed=124.445374ms filename=python.py language=
time="2018-02-26T10:23:03Z" level=warning msg="unable to allocate a driver instance: timeout, all drivers are busy" language=python
time="2018-02-26T10:23:03Z" level=error msg="request processed content 23 bytes, status Fatal" elapsed=5.00086396s filename=python.py language=python
```

--- a/README.md
+++ b/README.md
@@ -62,6 +62,36 @@ and an example contained in the docker image:
 docker exec -it bblfshd bblfshctl parse /opt/bblfsh/etc/examples/python.py
 ```
 
+## SELinux
+
+If your system has SELinux enabled (is the default in Fedora, Red Hat, CentOS
+and many others) you need to compile and load a policy module before running the
+bblfshd Docker image or running driver containers will fail with a `permission
+denied` message in the logs. 
+
+To do this, run these commands from the project root:
+
+```bash
+cd selinux/
+sh compile.sh
+semodule -i bblfshd.pp
+```
+
+If you were already running an instance of bblfshd, you will need to delete the
+container (`docker rm -f bblfshd`) and run it again (`docker run...`).
+
+Once the module has been loaded with `semodule` the change should persist even
+if you reboot. If you want to permanently remove this module run `semodule -d bblfshd`.
+
+Alternatively, you could set SELinux to permissive module with:
+
+```
+echo 1 > /sys/fs/selinux/enforce
+```
+
+(doing this on production systems which usually have SELinux enabled by default
+should be strongly discouraged).
+
 ## Development
 
 If you wish to work on *bblfshd* , you'll first need [Go](http://www.golang.org)

--- a/runtime/runtime.go
+++ b/runtime/runtime.go
@@ -196,8 +196,8 @@ func Bootstrap() {
 		factory, _ := libcontainer.New("")
 		if err := factory.StartInitialization(); err != nil {
 			panic("error bootstraping container " +
-			      "(hint: if SELinux is enabled, compile and load the policy module " +
-				  "in the selinux/ directory in the bblfshd repo): " + err)
+				"(hint: if SELinux is enabled, compile and load the policy module " +
+				"in this repo's selinux/ directory): " + err)
 		}
 		panic("--this line should have never been executed, congratulations--")
 	}

--- a/runtime/runtime.go
+++ b/runtime/runtime.go
@@ -195,7 +195,9 @@ func Bootstrap() {
 		runtime.LockOSThread()
 		factory, _ := libcontainer.New("")
 		if err := factory.StartInitialization(); err != nil {
-			panic(err)
+			panic("error bootstraping container " +
+			      "(hint: if SELinux is enabled, compile and load the policy module " +
+				  "in the selinux/ directory in the bblfshd repo): " + err)
 		}
 		panic("--this line should have never been executed, congratulations--")
 	}

--- a/runtime/runtime.go
+++ b/runtime/runtime.go
@@ -4,6 +4,7 @@ import (
 	"os"
 	"path/filepath"
 	"runtime"
+	"strings"
 	"syscall"
 
 	"github.com/opencontainers/runc/libcontainer"
@@ -195,9 +196,13 @@ func Bootstrap() {
 		runtime.LockOSThread()
 		factory, _ := libcontainer.New("")
 		if err := factory.StartInitialization(); err != nil {
-			panic("error bootstraping container " +
-				"(hint: if SELinux is enabled, compile and load the policy module " +
-				"in this repo's selinux/ directory): " + err)
+			if strings.Contains(err.Error(), "permission denied") {
+				panic("error bootstraping container " +
+					"(hint: if SELinux is enabled, compile and load the policy module " +
+					"in this repo's selinux/ directory): " + err.Error())
+			} else {
+				panic(err)
+			}
 		}
 		panic("--this line should have never been executed, congratulations--")
 	}

--- a/selinux/bblfshd.te
+++ b/selinux/bblfshd.te
@@ -1,0 +1,11 @@
+
+module bblfshd 1.0;
+
+require {
+        type container_runtime_t;
+        type spc_t;
+        class fifo_file setattr;
+}
+
+#============= spc_t ==============
+allow spc_t container_runtime_t:fifo_file setattr;

--- a/selinux/compile.sh
+++ b/selinux/compile.sh
@@ -1,0 +1,7 @@
+#!/bin/sh
+
+checkmodule -M -m -o bblfshd.mod bblfshd.te && \
+    semodule_package -o bblfshd.pp -m bblfshd.mod && \
+    echo 'Module compiled, load it with semodule -i bblfshd.pp'
+
+# enable with: semodule -i bblfshd.pp

--- a/selinux/compile.sh
+++ b/selinux/compile.sh
@@ -2,6 +2,4 @@
 
 checkmodule -M -m -o bblfshd.mod bblfshd.te && \
     semodule_package -o bblfshd.pp -m bblfshd.mod && \
-    echo 'Module compiled, load it with semodule -i bblfshd.pp'
-
-# enable with: semodule -i bblfshd.pp
+    echo 'Module compiled, load it with "semodule -i bblfshd.pp"'

